prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -114,8 +115,9 @@ def assign_substations_to_lines(substations, lines, line2sub):
             for e, e_sub in end_sub.items()
         ]
     )
+    tree = KDTree([ll2uv(p[0], p[1]) for p in subcoord])
```

Comment:
Shall we refactor `find_closest_neighbor` in this way so that any function calls `find_closest_neighbor` will benefit from the performance improvement?

Reply:
I think the performance improvement comes when we have many points for which we want to find the closest neighbor in another set of points. If we build the tree each time, we may not get any performance improvement.

Reply:
yes that is the problem. In fact I first refactored `find_closest_neighbor` to use `KDTree` but it was very slow because of the reason mentioned by @danielolsen, i.e., the tree was built for each iteration

Reply:
It could be nice to have a new function `find_closest_neighbors`, to replace any repetitive calls to `find_closest_neighbor` or more manual `haversine` loops.

Reply:
In fact we could write a new function that takes a list of coordinates that will be associated to a closest neighbor in a second list of points. It would return a list of id in the second list (neighbors).

Reply:
It would be a thin wrapper around converting lat/lon to unit vector via `ll2uv`, building a `KDTree`, and querying it. Simple but handy.

Reply:
Agree.